automerge-backend/src/backend.rs

```diff
@@ -354,6 +354,25 @@ impl Backend {
             .and_then(|index| self.history.get(*index))
     }
 
+    pub fn get_changes_added<'a>(&self, other: &'a Self) -> Vec<&'a Change> {
+        let mut stack: Vec<_> = other.op_set.deps.iter().collect();
+        let mut seen_hashes = HashSet::new();
+        let mut added_change_hashes = Vec::new();
```

Comment:
Do we need both `seen_hashes` and  `added_change_hashes`? They look like they are the same so we could just use the `seen_hashes`.

Reply:
From looking at the comments in the JS version, it seems like we want to preserve the order of the DFS.